ecdsa/src/recovery.rs

```diff
@@ -266,11 +266,31 @@ where
 
     /// Recover a [`VerifyingKey`] from the given `prehash` of a message, the
     /// signature over that prehashed message, and a [`RecoveryId`].
-    #[allow(non_snake_case)]
     pub fn recover_from_prehash(
         prehash: &[u8],
         signature: &Signature<C>,
         recovery_id: RecoveryId,
+    ) -> Result<Self> {
+        Self::recover_internal(prehash, signature, recovery_id, true)
+    }
+    /// Recover a [`VerifyingKey`] from the given `prehash` of a message, the
+    /// signature over that prehashed message, and a [`RecoveryId`]. Compared to
+    /// `recover_from_prehash`, this function skips verification with the
+    /// recovered key.
+    pub fn recover_from_prehash_noverify(
+        prehash: &[u8],
+        signature: &Signature<C>,
+        recovery_id: RecoveryId,
+    ) -> Result<Self> {
+        Self::recover_internal(prehash, signature, recovery_id, false)
+    }
+
+    #[allow(non_snake_case)]
+    fn recover_internal(
```

Comment:
Rather than adding a third function with a boolean `verify` flag, it seems like `recover_from_prehash` could just call `recover_from_prehash_noverify` and then verify the signature

Reply:
Sure. Another issue: `trial_recovery_from_prehash` can also benefit from `recover_from_prehash_noverify`. Can we update it too?

Reply:
I would prefer to leave that for a followup PR